pygbm/gradient_boosting.py

```diff
@@ -78,10 +92,10 @@ def fit(self, X, y):
 
         if self.verbose:
             print("Fitting gradient boosted rounds:")
-        # TODO: plug custom loss functions
-        y_pred = np.zeros_like(y_train, dtype=np.float32)
-        gradients = np.asarray(y_train, dtype=np.float32).copy()
-        hessians = np.ones(1, dtype=np.float32)
+        # TODO: is the initial prediction always 0? What about classif?
+        y_pred = np.zeros_like(y_train)
+        gradients, hessians = self.loss_.init_gradients_and_hessians(
+            n_samples=y_train.shape[0])
```

Comment:
Actually, it might even be smarter to combine the computation of the initial `gradients`, `hessians` and baseline `y_pred` in a single method as it's likely that they can share some intermediate computation.

```python
        y_pred, gradients, hessians = self.loss_.baseline_predictions(y_train)
```

Again, this should better be addressed in a separate PR.